web/src/components/organisms/Project/Content/ContentDetails/hooks.ts

```diff
@@ -0,0 +1,100 @@
+import { useCallback, useMemo } from "react";
+
+import { FieldType } from "@reearth-cms/components/molecules/Schema/types";
+import {
+  SchemaFiledType,
+  useCreateItemMutation,
+  useUpdateItemMutation,
+} from "@reearth-cms/gql/graphql-client-api";
+
+import useContentHooks from "../hooks";
+
+type Props = {
+  itemId: string | undefined;
+};
+
+export default ({ itemId: itemID }: Props) => {
+  const { currentModel, itemsData } = useContentHooks();
```

Comment:
It looks like you are setting currentModel in Jotai state. Why don't you import in `useModel` here instead of using another component's hooks. 
As for itemsData, is it not desirable to hold that in Jotai state as well ?

Reply:
@KaWaite , the **currentModel** is set in the `modelMenu` organism (hook) and components with no direct relation to the modelsMenu component use its state and a couple of components share the state.

On the other hand, `ItemsData` state is needed on a couple of screens which has a direct relationship with the items (content table and content form), and the two pages are using the same query, so query caching does the trick to not load the **graphQL** more than once.

Reply:
@KaWaite , the **currentModel** is set in the `modelMenu` organism (hook) and components with no direct relation to the modelsMenu component use its state and a couple of components share the state.

On the other hand, `ItemsData` state is needed on a couple of screens which has a direct relationship with the items (content table and content form), and the two pages are using the same query, so query caching does the trick to not load the **graphQL** more than once.

Reply:
Ok, for now this is fine. But I do want to discuss with you and other FE members soon about "global" or "shared" hooks files. For a long time with reearth-web we have dealt with duplicate hooks and its really not nice.